http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -912,6 +941,10 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       else Future.successful(Map.empty[String, String])
       clusterVPCSettings = getClusterVPCSettings(projectLabels)
 
+      // If user is using https://github.com/DataBiosphere/terra-docker/tree/master#terra-base-images for jupyter image, then
+      // we will use the new custom dataproc image
+      dataprocImage = if(clusterImages.exists(_.dockerImage.contains("terra-jupyter"))) dataprocConfig.customDataprocImage else dataprocConfig.legacyCustomDataprocImage
```

Comment:
don't the terra-docker hail images 0.1 and 0.2 depend on old spark?

Reply:
I don’t think hail 0.1 and 0.2 actually worked. So i think we can assume 0.1 and 0.2 should never be used